### ISSUE TITLE
What is the proper way to create repository instances?

### ISSUE THREAD

**Issue by welkie** [IRCraziestTaxi/typeorm-linq-repository#35]
I noticed that in TypeORM's documentation, they recommend you call `getRepository` on the connection after it has been set up. This seems like a nice way to do DI, so you can repository instances connected to different databases if that's what you want.

As I was testing this library, I created some unit tests and noticed the way I was creating the repositories seems to hide a bit of magic. Since the documentation for this library instructs us to make our own repository classes, the only way I could think of to get an instance of them would be to call their constructors. But that would mean not specifying which connection should be used for them:

```typescript
import { createConnection, Connection } from 'typeorm';

import { typeOrmConfigTest as typeOrmConfig } from '../src/config';

import { DoctorRepository } from '../src/models/repositories/DoctorRepository';
import { PatientRepository } from '../src/models/repositories/PatientRepository';
import { AppointmentRepository } from '../src/models/repositories/AppointmentRepository';

describe('Database', async function () {
    this.timeout(10000);

    let conn: Connection;

    let patientRep: PatientRepository;
    let doctorRep: DoctorRepository;
    let appointmentRep: AppointmentRepository;

    // Set up connection and repositories for tests
    before(async () => {
        conn = await createConnection(typeOrmConfig);

        patientRep = new PatientRepository();
        doctorRep = new DoctorRepository();
        appointmentRep = new AppointmentRepository();
    });

    // tests here...
}
```

Is this the right approach, or should we be getting instances of our repositories another way?

**Comment by IRCraziestTaxi** [comment]
I actually thought recently about this problem - the ability to create repositories from different connections. So I'm glad it came up now. It should be trivial to implement. I imagine the end result will look something like:

```ts
// Create repository from single or default connection (current implementation).
class EntityRepository extends RepositoryBase<Entity> {
    public constructor() {
        super(Entity);
    }
}

// Pass optional connection name to get into typeorm-linq-repository's base repository constructor.
class EntityRepository extends RepositoryBase<Entity> {
    public constructor() {
        super(Entity, true, "connection-name");
    }
}
```

I will take care of this soon and will let you know when it is done.

Thanks!

**Comment by IRCraziestTaxi** [comment]
Fixed in `1.0.0-alpha.8`.

```ts
class EntityRepository extends RepositoryBase<Entity> {
    public constructor(connectionName: string) {
        super(Entity, connectionName);
    }
}

entityRepo = new EntityRepository("connection-name");
```

**Comment by welkie** [comment]
@IRCraziestTaxi Nice!

I'm not super knowledgeable with how TypeORM uses constructors under the hood, but regarding this syntax:

```typescript
super(Entity, connectionName)
```

Have you considered, if possible, using an object with a params type instead, for more readable code and to support more future features? Like:

```typescript
super(Entity, {
  connectionName: "abc123",
});
```

**Comment by IRCraziestTaxi** [comment]
@welkie Good call. Implemented in `1.0.0-alpha.9`.
